src/Monolog/Processor/RedactSensitiveProcessor.php

```diff
@@ -0,0 +1,77 @@
+<?php
+declare(strict_types=1);
+
+namespace Monolog\Processor;
+
+use Monolog\LogRecord;
+
+final class RedactSensitiveProcessor implements ProcessorInterface
+{
+    /** @var string[] */
+    private array $sensitiveKeys;
+
+    /** @var string[] */
+    private array $patterns;
+
+    private string $mask;
+
+    /**
+     * @param string[] $sensitiveKeys exact keys to redact in context/extra (case-insensitive)
+     * @param string[] $patterns PCRE regex patterns to mask inside string values (e.g. '/Bearer\\s+[A-Za-z0-9\\._-]+/')
+     * @param string   $mask replacement token
+     */
+    public function __construct(array $sensitiveKeys = ['password','passwd','pwd','secret','token','api_key','apikey','authorization','auth','cookie'], array $patterns = [], string $mask = 'REDACTED')
```

Comment:
Just an idea.. not sure if good or bad, but this should catch any long word of 30+ [a-z0-9] characters (optionally prefixed by foo_bar_ or the like, like many API tokens are), which is very likely to be a secret/token.. But of course it risks some collateral damage.

```suggestion
    public function __construct(array $sensitiveKeys = ['password','passwd','pwd','secret','token','api_key','apikey','authorization','auth','cookie'], array $patterns = ['{\b(?:[a-z]+_)*[a-zA-Z0-9]{30,}\b}'], string $mask = 'REDACTED')
```